packages/api/e2e/amino.spec.ts

```diff
@@ -96,35 +95,9 @@ describe('RegenApi with tendermint connection', () => {
       let txRes: DeliverTxResponse | undefined;
       const api = await connect();
       const msgClient = api.msgClient;
-      const mnemonic =
-        'time dice choose cabbage suit panic silly cattle picture auto grab hole'; //regen1m0qh5y4ejkz3l5n6jlrntxcqx9r0x9xjv4vpcp
-
-      const signer = await Secp256k1HdWallet.fromMnemonic(mnemonic, {
-        prefix: 'regen',
-      });
-
-      const signDoc = makeSignDoc(
-        [{ type: MsgSend.$type, value: TEST_MSG_SEND }],
-        TEST_FEE,
-        'regen-redwood-1',
-        TEST_MEMO,
-        96,
-        2, //regen q auth account regen1m0qh5y4ejkz3l5n6jlrntxcqx9r0x9xjv4vpcp
-      );
-
-      const { signature, signed } = await signer?.signAmino(
-        TEST_ADDRESS,
-        signDoc,
-      );
-
-      const signedTxBytes = decodeSignature(signature).signature;
-      // const signedTxBytes = await msgClient?.sign(
-      //   TEST_ADDRESS,
-      //   [TEST_MSG_SEND],
-      //   TEST_FEE,
-      //   TEST_MEMO,
-      // );
 
+      const signedTxBytes = await msgClient?.sign(TEST_ADDRESS,[TEST_MSG_SEND],TEST_FEE,TEST_MEMO)
```

Comment:
@mhagel I discovered this `sign` function on the msgClient, added in regen-js, which has a nice API.

It handles both amino and proto. Since the signer is an amino signer, here we can be sure that amino is used behind the scenes. Your `'toAmino sender,recipient,credits,'` logs also show up.

Reply:
I however have another (more friendly) error now:

```
Broadcasting transaction failed with code 4 (codespace: sdk). Log: signature verification failed; please verify account number (96), sequence (4) and chain-id (regen-redwood-1): unauthorized
```

Is `regen-redwood-1` the correct chain-id?

Reply:
>It handles both amino and proto

That's what I thought, but wasn't sure. Thanks.

>`Log: signature verification failed`

Yes, I ran into that error for a while. That was the auth error I was seeing for a while. Not sure if I got past it, or just ran into a different error eventually. I thought it could be a sequence mismatch or something. The chain-id should be right.

Reply:
After debugging with a local node, I think I found the signature mismatch bug. It's actually a regen-ledger bug.

Here's what the JS code signs:

```json
{"account_number":"7","chain_id":"my-chain","fee":{"amount":[{"amount":"5000","denom":"uregen"}],"gas":"200000"},"memo":"regen-js v0.1.0 tests","msgs":[{"type":"regen/MsgSend","value":{"credits":[{"batch_denom":"C01-001-20170606-20210601-007","tradable_amount":"0.01"}],"recipient":"regen1m0qh5y4ejkz3l5n6jlrntxcqx9r0x9xjv4vpcp","sender":"regen1m0qh5y4ejkz3l5n6jlrntxcqx9r0x9xjv4vpcp"}}],"sequence":"0"}
```

here's what the node expects to be signed:

```json
{"account_number":"7","chain_id":"my-chain","fee":{"amount":[{"amount":"5000","denom":"uregen"}],"gas":"200000"},"memo":"regen-js v0.1.0 tests","msgs":[{"type":"regen/MsgSend","value":{"credits":[{"batch_denom":"C01-001-20170606-20210601-007","tradable_amount":"0.01"}],"recipient":"regen1m0qh5y4ejkz3l5n6jlrntxcqx9r0x9xjv4vpcp","sender":"regen1m0qh5y4ejkz3l5n6jlrntxcqx9r0x9xjv4vpcp"}}],"sequence":"0"}
```

Notice how JS adds `type` and `value` subfields in the msgs array, which is correct. This bug also happened on the SDK once before: https://medium.com/confio/authz-and-ledger-signing-an-executive-summary-8754a0dc2a88.

I created PR to show this: https://github.com/regen-network/regen-ledger/pull/1480. With that PR and a local node, I do not get any sig mismatch error.